source/extensions/clusters/strict_dns/strict_dns_cluster.cc

```diff
@@ -227,14 +233,40 @@ void StrictDnsClusterImpl::ResolveTarget::startResolve() {
       });
 }
 
+absl::StatusOr<Network::DnsResolverSharedPtr> StrictDnsClusterFactory::selectDnsResolver(
```

Comment:
Its awkward regardless of how its done because `selectDnsResolver(cluster, context)` is a method. So the other options are to

* write a separate *function* (not method) to share the code. This is awkward because then half the implementation is in the class and the other is outside.
* add another level of inheritance to the factories to share the method.

Given that ideally, these factory implementations are eventually going to be merged together, I chose to go with the most straightforward method and duplicate code, but let me know what you think.

I will also add that I think the inheritance hierarchy should really be flattened.

Reply:
Maybe you could just move this implementation into the base class without creating a new level in the hierarchy?

Similarly like the original selectDnsResolver is already a method in a base class, but with different paramters. Though it's just a nitpick, so I would not insist anyways.